Alarmierungen in einen Kalender eintragen

nobbie2009 opened the pull request:
Gleich vorweg, es ist mein erstes Plugin, es ist mein erster PullRequest .. bitte um Nachsicht ;)

Dieses Plugin, macht wie der Titel schon sagt nichts anderes als eine Alarmierung in eine ICS-File zu schreiben. 
Benötigt werden hierfür noch folgende Projekte:

icalendar https://pypi.org/project/icalendar/
dateutil https://github.com/dateutil/dateutil/
pytz https://pypi.org/project/pytz/

nobbie2009 committed:
Add files via upload

Auslagerung der Filter aus der normalen Config

nobbie2009 committed:
Update config.template.ini

Filter ausgelagert in config_filter.ini

nobbie2009 committed:
Update boswatch.py

-Auslagerung der Filter in 2. Datei
-hinzufügen einer globalen Variable für den Testbetrieb - Dadurch können Filter getestet werden ohne diese ständig eingeschaltet zu haben

nobbie2009 committed:
Create 2calendar.py

Schreibt Alarme in eine ICS-Datei

nobbie2009 committed:
Update config.template.ini

Config für 2Calendar hinzugefügt

nobbie2009 committed:
Update 2calendar.py

Überprüfung ob Kalender bereits existiert, falls ja, werden vorhandene Einträge importiert.

nobbie2009 committed:
Update 2calendar.py

flothi commented:
Schaut auf den ersten Blick soweit gut aus, allerdings gefällt mir der hardgecodete Pfad in der boswatch.py nicht
Weiterhin wäre es natürlich super, das Plugin direkt um die Alarmierung mittels POCSAG zu ergänzen und nicht auf ZVEI zu beschränken.

nobbie2009 committed:
Update boswatch.py

Pfad "softgecoded" ;)
hatte es im html Ordner für individuelle Anpassung via php

nobbie2009 commented:
POCSAG habe ich keinen Bezug hier bei uns läuft alles noch im "guten" !altem! ZVEI 

PS: Pfad habe ich angepasst.

Schrolli91 commented:
Bitte die merge Konflikte noch beheben vor einem ausführlichen Review (aktuellen develop in deinen Entwicklungsbranch mergen und die Fehler beseitigen)

Schrolli91 commented on boswatch.py:
```diff
@@ -156,7 +156,9 @@
 		if args.test:
 			logging.debug(" - Test-Mode!")
-
+			globalVars.testvariable = 1
+		else:
+			globalVars.testvariable = 0
```
Was ist das? Wofür wird das benötigt?

nobbie2009 replied:
Das ist eine Testvariable, mit der ich unter anderem in meinem Testbetrieb geschaut hab, ob Boswatch im Testmodus lief, somit konnte ich die Probealarmierungen abfangen und den Start anderer Plugins unterbinden, kann aber gern weg.

Schrolli91 commented on boswatch.py:
```diff
@@ -202,7 +204,7 @@
 	try:
 		logging.debug("reading config file")
 		globals.config = ConfigParser.ConfigParser()
-		globals.config.read(globals.script_path+"/config/config.ini")
+		globalVars.config.read([globalVars.script_path+"/config/config.ini", "globalVars.script_path+"/config/config_filter.ini"])
```
Wieso wurde die Konfig ausgelagert?

nobbie2009 replied:
hehe, hat indirekt auch nichts mit dem Plugin zu tun, habe ich gemacht um die Filter direkt per PHP ändern zu können ohne in die Config selbst reinzuschreiben.. 

Merke: für das nächste mal nicht alles in einen PR ;)

Schrolli91 commented on plugins/2calendar/2calendar.py:
Debug Ausgaben sollten entfernt werden

nobbie2009 replied:
ok

Schrolli91 reviewed (changes requested):
Ein paar Kleinigkeiten sind mir aber vorab schon aufgefallen ...

Schrolli91 commented:
@nobbie2009 das schließen wäre nicht notwendig gewesen - man kann einen PR auch nachträglich verändern
